### DISCUSSION TITLE
How to set page schema markup (JSON-LD) based on data from API call?

### DISCUSSION THREAD

**Discussion by Sandlens** [nuxt/nuxt#16748] [General]
I am inserting schema markup into some pages via a JSON-LD script:

```
<Head>
  <Script type="application/ld+json" :children="JSON.stringify(schema)"></Script>
</Head>
```

After making an API call to my backend (Firebase) and receiving data back, I would like to add this data to the `schema` object. The `<Script>` is currently loading before the API call is complete, so the schema is therefore incomplete.

Is there some way to make the `<Script>` `await` the completion of the API call? I have tried `useHead` but haven't gotten it working either. Would appreciate any advice. Thank you.

**Comment by danielroe** [comment]
You can `await` the result of the fetch within your setup function.

**Reply by Sandlens** [reply]
Thank you very much!
